### DISCUSSION TITLE
Dashboard improvements

### DISCUSSION THREAD

**Discussion by GWeidmann** [datalust/seq-tickets#1980] [Ideas]
Dashboards are quite useful to get a quick look, but there is a thing that I am missing and a thing that bothers me:

## First the bothering thing: The chart colors.

![image](https://github.com/datalust/seq-tickets/assets/97590580/79dab0f4-9992-4d14-99aa-da352337f7e4)

I got a chart on my dashboard for every single app to get a quick look if there are any warnings, errors, or even fatals occuring. But the colors are not consistent and therefore it's difficult to get a quick overview. Maybe it would be possible to assing the colors according to a hash value of the string? Then it would be always the same color for the same value.

## Improvement: Placeholders

The other thing that would help me are placeholders or something like that. At the moment I lay out my dashboard to keep similar charts together. But as the layout fills empty gaps automatically I need to do this at the moment:

![image](https://github.com/datalust/seq-tickets/assets/97590580/ec6d5de1-3f62-4d1e-a782-2c639ada7f63)

It would be helpful if I didn't had to insert an empty chart (with custom SQL condition `1 = 0`)

**Comment by liammclennan** [comment]
Hi @GWeidmann 

Thanks for the ideas. The colors one is tricky. For example, consider the scenario where 'Warning', 'Information' and 'Error' all hash to the same color. We need to think about that some more.

**Reply by GWeidmann** [reply]
Yes I understand, was just a suggestion. Maybe also setting custom default colors globally would be an option. The important thing is just that they stay the same. (even if custom colors would be great, to make e.g. warnings yellows, warnings orange, fatals red

**Comment by Oscar-Santacruz** [comment]
It would be highly beneficial to introduce a customization option for HTTP status code colors in Seq, with a simple rule to prevent colors from being repeated across codes.  

This feature is crucial for maintaining clarity in monitoring and reporting. Currently, we experience situations where the same HTTP status code, such as `200`, is displayed with different colors (e.g., green in one instance and red in another). This inconsistency creates confusion for both the monitoring team and stakeholders, especially during quick assessments or reports.  

Having a way to assign specific, consistent colors to HTTP codes would improve usability and make dashboards more intuitive.  

**Proposed Implementation:**  
- Allow users to configure default colors for common HTTP codes (e.g., 200, 404, 500).  
- Enforce a rule to prevent duplicate colors for different codes.  
- Optionally provide a fallback to assign auto-generated distinct colors for any codes not explicitly defined by the user.  

This enhancement would greatly enhance Seq's effectiveness in real-time monitoring scenarios.
